### PR TITLE
Updated: Vendors list page

### LINKED ISSUE #845: More responsive and filterable vendors list
**Is your feature request related to a problem? Please describe.**
Currently, The vendor lists are shown as 2 columns on desktop and 1 on mobile and we can not filter vendors
image

**Describe the solution you'd like**
Make vendor lists more responsive and filterable. eg filter by best selling vendor or most rating. We need to display vendor lists as a grid or list. Vendor lists should be shown in at least 3 columns on the desktop and 2 columns on mobile or one on the small screen
**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by Mrlunsuper**
### All Submissions:

* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #845.
- Redesign vendors list with two display mode and responsive
- Added option to select default avatar source and display mode `WC Vendors -> Settings -> Display`
- Added vendor search bar
- Show more vendor information

### How to test the changes in this Pull Request:

1. Go to vendors page
2. Try to search and change display mode

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

### CODE REVIEW

File: classes/admin/settings/class-wcv-settings-display.php [hunk 1]
```diff
@@ -315,12 +325,41 @@ public function get_settings( $current_section = '' ) {
...
+							'id'      => 'wcvendors_display_vendors_list_type',
+							'default' => 'grid',
+							'type'    => 'select',
+							'class'   => 'wc-enhanced-select',
+							'options' => array(
+								'grid' => __( 'Grid', 'wc-vendors' ),
+								'list' => __( 'List', 'wc-vendors' ),
+							),
+						),
+						array(
```
Comment (digitalchild) [inline classes/admin/settings/class-wcv-settings-display.php:350]: We should use the avatar options available in WordPress > Settings > Discussion to give greater flexibility for avatar displays.

**Review by digitalchild — CHANGES_REQUESTED** [review]
This is looking great, see my few comments for updates.

**Review by mahlamusa — APPROVED** [review]

**Review by ketulsheth85 — APPROVED** [review]

File: templates/front/vendor-list-loop.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+ *  $address : the vendor store address
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+?>
+<div class="vendor_list">
+	<div class="vendor_list_avatar">
+		<?php echo $avatar; ?>
```
Comment (ketulsheth85) [inline templates/front/vendor-list-loop.php:28]: <a href="<?php echo esc_url( $shop_link ); ?>"><?php echo $avatar; ?></a>

Add hyperlink to the image

File: templates/front/vendor-list-loop.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	</div>
+	<div class="vendor_list_info">
+		<h3 class="vendor_list--shop-name"><?php echo esc_html( $shop_name ); ?></h3>
```
Comment (ketulsheth85) [inline templates/front/vendor-list-loop.php:31]: Add hyperlink here also same as above code.

**Review by digitalchild — APPROVED** [review]

**Pull request merged**